### DISCUSSION TITLE
Could Mad-x create the Twiss file with rotation alignment error?

### DISCUSSION THREAD

**Discussion by LiarGC** [MethodicalAcceleratorDesign/MAD-X#1100] [General]
I tried  to add an alignment error to a sextupole.
Firstly I gave it a position alignment error:
`EALIGN, DS := 5e-4*TGAUSS(2);`
The "L" of this sextupole in the Twiss file changed correctly.

But when adding the rotation misalignment:
`EALIGN,  DS := 5e-4*TGAUSS(2), DPSI := 5e-4*TGAUSS(2);`
The "K2SL" of this sextupole in the Twiss file was 0, while the "K2L" was still its original value.

I saved and checked the error file, finding no problem with the rotation misalignment. And I tried the PTC_TWISS function, which gave the same result, "K2SL"=0.

Did I miss some key word to include the rotation misalignment?

**Comment by LiarGC** [comment]
There is one mistake, The "L" didn't change either, only the twiss parameters, such as BetaX, BetaY, changed.

**Comment by rdemaria** [comment]
Twiss stable contains the attributes of the element without taking into account the imperfections defined by EALIGN or EFCOMP. The imperfections can be retrieved using the ESAVE, but also in this case you will get rotations and multipoles error separated.
